### PR TITLE
Add bool func to return if creature has health

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Add hasHealth function instead comparing creature health

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -4156,7 +4156,7 @@ bool Game::combatChangeHealth(Creature* attacker, Creature* target, CombatDamage
 {
 	const Position& targetPos = target->getPosition();
 	if (damage.primary.value > 0) {
-		if (target->getHealth() <= 0) {
+		if (!target->hasHealth()) {
```
Comment (marmichalski) [inline src/game.cpp:4159]: no, no, such quality of life improvement should be accompanied by `isDead` function

Comment (ramon-bernardo) [reply src/game.cpp:4159]: isDead?

Comment (marmichalski) [reply src/game.cpp:4159]: Every creature has health. The only thing that you can assert on it is whether it is alive (>0) or dead.

Comment (ramon-bernardo) [reply src/game.cpp]: Thanks!

File: src/spells.cpp [hunk 2]
```diff
@@ -850,7 +850,7 @@ bool InstantSpell::playerCastInstant(Player* player, std::string& param)
 			}
 
 			target = playerTarget;
-			if (!target || target->getHealth() <= 0) {
+			if (!target || target->isDead()) {
```
Comment (ramon-bernardo) [inline src/spells.cpp:853]:
```suggestion
			if (!target || target->isRemoved() || target->isDead()) {
```
Missing removed check?

File: src/spells.cpp [hunk 3]
```diff
@@ -884,7 +884,7 @@ bool InstantSpell::playerCastInstant(Player* player, std::string& param)
 			}
 		} else {
 			target = player->getAttackedCreature();
-			if (!target || target->getHealth() <= 0) {
+			if (!target || target->isDead()) {
```
Comment (ramon-bernardo) [inline src/spells.cpp:887]: Same here

**Comment by yamaken93** [comment]
Maybe has hasHealth() could be reaplced with !isDead()? I believe in the contexts that hasHealth() is called we are checking to see if the creature is alive (!isdead).

**Comment by ramon-bernardo** [comment]
> Maybe has hasHealth() could be reaplced with !isDead()? I believe in the contexts that hasHealth() is called we are checking to see if the creature is alive (!isdead).

isAlive and isDead?

**Comment by EPuncker** [comment]
> > Maybe has hasHealth() could be reaplced with !isDead()? I believe in the contexts that hasHealth() is called we are checking to see if the creature is alive (!isdead).
> 
> isAlive and isDead?

isDead make more sense

**Review by EPuncker — APPROVED** [review]

**Pull request merged**